cmd/minishift/cmd/start.go

```diff
@@ -544,11 +544,14 @@ func registerHost(libMachineClient *libmachine.Client) {
 		log.Debug("Skipping registration due to enabled --skip-registration flag")
 		return
 	}
-	if err := cluster.Register(libMachineClient); err != nil {
+	supportRegistration, err := cluster.Register(libMachineClient)
+	if err != nil {
 		atexit.ExitWithMessage(1, fmt.Sprintf("Error to register VM: %v", err))
-	} else {
+	}
+	if supportRegistration {
 		minishiftConfig.InstanceConfig.IsRegistered = true
 		minishiftConfig.InstanceConfig.Write()
+		return
```

Comment:
@LalatenduMohanty we shouldn't because cluster.go is part of minikube pkg and moving forward we are thinking to consume it as library so moving this code doesn't make sense.